framework/src/st_things/things_stack/cloud/cloud_manager.c

```diff
@@ -1571,15 +1573,25 @@ static char *make_cloud_address(char *ip, char *port, const char *ci_addr)
 	// Update Cloud Address.
 	memset(g_cloud_address, 0, MAX_CI_ADDRESS);
 	if (things_strcat(g_cloud_address, MAX_CI_ADDRESS, DEFAULT_COAP_TCP_HOST) == NULL) {
+		if (is_ci_addr_null) {
+			things_free(ipport);
```

Comment:
Why is it freed here? ipport is assigned 1560 as a parameter.
My thinking, ci_addr will free when returns NULL.

Reply:
As we are returning from the function, and we dont require `ipport` since `ci_addr` is NULL, so, we can free it.

Reply:
@sushant-23 Right. But It looks not good. 
You'd better it frees in caller function.
Any function or module has allocated a memory then the function or module will free the memory.

Reply:
@Hyunjun85, 
Thanks for approving.
I would like to clarify. We have to de-allocate memory which was allocated @ line: 1569 before returning. This memory allocation is done only when 'ci_addr' which is passed by the caller is NULL. So the 'if' block will be executed only when the memory has been allocated.